### PR TITLE
Part of mandatory tests

### PR DESCRIPTION
**Pull request by domachine**
This PR implements some of the mandatory tests mentioned in #36.

These tests were added:

- "6.1.16"
- "6.1.17"
- "6.1.18"
- "6.1.19"
- "6.1.20"
- "6.1.22"

### CODE REVIEW

File: app/lib/shared/Core/entities/DocumentEntity.js [hunk 1]
```diff
@@ -483,6 +486,115 @@ export default class DocumentEntity {
...
+      hasTrackingRevisionHistory(doc) &&
+      hasTrackingVersionField(doc) &&
+      doc.document.tracking.revision_history.length > 0
+    ) {
+      if (
+        doc.document.tracking.revision_history
+          .slice()
+          .sort(
+            (a, z) => new Date(z.date).getTime() - new Date(a.date).getTime()
+          )[0].number !== doc.document.tracking.version
```
Comment (tschmidtb51) [inline app/lib/shared/Core/entities/DocumentEntity.js:499]: According to [CSAF 6.1.6](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.0/prose/csaf-v2-editor-draft.md#6116-latest-document-version) build metadata is ignored in the comparison.

File: app/tests/shared/coreFixture.js [hunk 2]
```diff
@@ -418,5 +418,32 @@ export default {
...
+                date: '2021-07-21T10:00:00.000Z',
+                number: '1',
+                summary: 'Some other changes.',
+              },
+            ],
+            version: '1',
+          },
+        },
+      },
+    },
```
Comment (tschmidtb51) [inline app/tests/shared/coreFixture.js:447]: I suggest to add an additional test with semantic versioning here:
```suggestion
    },

    // Fails "6.1.22 Multiple Definition in Revision History"
    {
      valid: false,
      content: {
        ...MINIMAL_DOC,
        document: {
          ...MINIMAL_DOC.document,
          tracking: {
            ...MINIMAL_DOC.document.tracking,
            revision_history: [
              {
                date: '2021-07-20T10:00:00.000Z',
                number: '1.0.0',
                summary: 'Initial version.',
              },
              {
                date: '2021-07-21T10:00:00.000Z',
                number: '1.0.0',
                summary: 'Some other changes.',
              },
            ],
            version: '1.0.0',
          },
        },
      },
    },

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
@domachine Thanks for implementing the tests.

I have some requests for changes. Please address the comments.

FYI: Also I opened tickets (https://github.com/oasis-tcs/csaf/issues/339, https://github.com/oasis-tcs/csaf/issues/340) for parts of the spec to remove ambiguity.

**Comment by tschmidtb51** [comment]
@domachine I'm sorry about the delayed review.

**Comment by domachine** [comment]
@tschmidtb51 No problem 🙂 . I addressed the issues. Could you re-check?

File: app/lib/shared/Core/entities/DocumentEntity.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply app/lib/shared/Core/entities/DocumentEntity.js]: We will keep track of https://github.com/oasis-tcs/csaf/issues/339 in https://github.com/secvisogram/secvisogram/issues/190.

**Review by tschmidtb51 — APPROVED** [review]
LGTM.

**Review by pdamian — APPROVED** [review]

**Review by bentolor — APPROVED** [review]

**Pull request merged**